pytorch_lightning/plugins/environments/slurm_environment.py

```diff
@@ -92,11 +92,10 @@ def job_name() -> Optional[str]:
     @staticmethod
     def job_id() -> Optional[int]:
         job_id = os.environ.get("SLURM_JOB_ID")
-        if job_id:
-            try:
-                job_id = int(job_id)
-            except ValueError:
-                job_id = None
+        try:
+            job_id = int(job_id)
+        except ValueError:
+            job_id = None
```

Comment:
1. you could return early if `job_id is None` to avoid this try/catch...
2. you could move the check from L101 to happen first. if the job_name is `"bash"` then also return `None` early. 
```suggestion
        if job_id is None:
            return None
        try:
            job_id = int(job_id)
        except ValueError:
            job_id = None
```

Reply:
Thanks for the feedback! I've implemented these changes in the new "change order" commit.